### PR TITLE
Filestore tweaks and improvements

### PR DESCRIPTION
**Pull request by derekcollison**
Optimized for startup restore time.

Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/filestore.go [hunk 1]
```diff
@@ -1586,54 +1593,42 @@ func (fs *fileStore) recoverFullState() (rerr error) {
...
-
 	// First let's check the happy path, open the blk file that was the lmb when we created the full state.
 	// See if we have the last block available.
 	var matched bool
 	var mb *msgBlock
 	if mb = fs.bim[blkIndex]; mb != nil {
-		matched = bytes.Equal(mb.lastChecksum(), lchk[:])
-		if matched && blkIndex == blkFiles.maxIndex {
-			return nil
+		if _, err := os.Stat(mb.mfn); err != nil && os.IsNotExist(err) {
```
Comment (neilalexander) [inline server/filestore.go:1604]: Is it safe to drop other `err` here?

File: server/filestore.go [hunk 2]
```diff
@@ -1586,54 +1593,42 @@ func (fs *fileStore) recoverFullState() (rerr error) {
...
-	start, stop := blkIndex, blkFiles.maxIndex
+	// We may need to check other blocks. Even if we matched last checksum we will see if there is another block.
+	// If we did not match we re-process the last block.
+	start := blkIndex
 	if matched {
 		start++
 	}
 
-	for bi := start; bi <= stop; bi++ {
+	for bi := start; ; bi++ {
```
Comment (neilalexander) [inline server/filestore.go:1625]: Minor nit but `for bi := start; true; bi++ {` might be clearer.

Comment (derekcollison) [reply server/filestore.go:1625]: I believe that is the way Go recommends it so will leave as is.

**Review by neilalexander — APPROVED** [review]
LGTM

**Pull request merged**